client/boot/common.js

```diff
@@ -88,6 +88,15 @@ const loggedOutMiddleware = currentUser => {
 	);
 
 	page( '*', ( context, next ) => {
+		if ( '/' === context.pathname && config.isEnabled( 'devdocs/redirect-loggedout-homepage' ) ) {
```

Comment:
Yes, I didn't think too much about it as it is enabled only on development envs.